### PR TITLE
DAAL covariance multi-gpu multi-node oneCCL sample

### PR DESCRIPTION
**Pull request by mchernov-intel**

### CODE REVIEW

File: samples/daal/cpp/oneccl/sources/error_handling.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    exit(fileError);
+}
+
+void sparceFileReadError()
+{
+    std::cout << "Incorrect format of file" << std::endl;
+    exit(fileError);
+}
+
+#endif
```
Comment (KulikovNikita) [inline samples/daal/cpp/oneccl/sources/error_handling.h:64]: EOL miss

File: samples/daal/cpp/oneccl/sources/service.h [hunk 2]
```diff
@@ -0,0 +1,804 @@
...
+SerializationIfacePtr deserializeDAALObject(daal::byte * buff, size_t length)
+{
+    /* Create a data archive to deserialize the object */
+    OutputDataArchive dataArch(buff, length);
+
+    /* Deserialize the numeric table from the data archive */
+    return dataArch.getAsSharedPtr();
+}
+
+#endif
```
Comment (KulikovNikita) [inline samples/daal/cpp/oneccl/sources/service.h:804]: Please check

File: samples/daal/cpp/oneccl/makefile_lnx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+$(RES_DIR)/%.exe: ./sources/%.cpp | $(RES_DIR)/.
+	$(CC) $(COPTS) $< -o $@ $(LOPTS)
+
+$(RES_DIR)/%.res: $(RES_DIR)/%.exe
+	$(CRUN) $(RUNOPTS) $< > $@
+
+#$(RES_DIR)/%.res:  $(RES_DIR)/%.exe
+#	$< > $@
+
+%/.:; mkdir -p $*
```
Comment (KulikovNikita) [inline samples/daal/cpp/oneccl/makefile_lnx:136]: EOL

File: samples/daal/cpp/oneccl/daal_lnx.lst [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+##  Content:
+##     Intel(R) oneAPI Data Analytics Library samples list
+##******************************************************************************
+
+ONECCL  =  kmeans_dense_distributed_oneccl
```
Comment (KulikovNikita) [inline samples/daal/cpp/oneccl/daal_lnx.lst:21]: EOL

**Review by KulikovNikita — APPROVED** [review]

**Comment by mchernov-intel** [comment]
moved to #1297

**Pull request closed without merging**